proposals/052-cryptography-leg-1.md

```diff
@@ -0,0 +1,329 @@
+
+### botan-bindings
+
+The `botan-bindings` library will contain raw bindings and is an almost direct, 1-1 translation of the C API into Haskell FFI calls. As such, it exposes and operates over C FFI types, requires buffer and pointer and pointer management, and returns error codes. This library only exposes FFI calls and constants, and is suitable for building your own abstraction over Botan.
+
+### botan-low
+
+The `botan-low` library will contain low-level bindings which wrap the FFI calls into monadic IO actions. This library handles the translation between buffers and ByteStrings, and throws exceptions in the case of errors, but is otherwise a fairly faithful translation of the interface. Thus this library is suitable for use, but is not recommended for day-to-day programming due to statefulness and a lack of type safety.
+
+### botan
+
+The `botan` library will contain high-level bindings which hides the stateful IO and exposes appropriately idiomatic and pure interfaces. It will provide data types for algorithms, enumerations for constants, convenience methods for key and nonce generation, incremental processing of lazy bytestrings, and otherwise bring significant safety and ergonomics over `botan-low` and `botan-bindings`.
+
+## Optional / Future Technical Content
+
+More advanced concepts such as typeclasses may be delivered in a future leg focused on generalization, but some development occur earlier in order to provide a testable 'use case' for the stack of libraries. If the first leg is completed ahead of schedule, the remaining effort will go towards these libraries.
+
+### crypto-schemes
+
+The `crypto-schemes` library will contain a hierarchy of abstract type classes and data families for cryptography and scheme identifiers, allowing for a unified interface to multiple back-ends. It will also provide various utility functions including sized bit and byte vectors, block and padding functions.
+
+### crypto-schemes-botan
+
+The `crypto-schemes-botan` library will be an implementation of `crypto-schemes` using `botan`, implementing class instances for the specific algorithms that `botan` provides.
+
+### botanium
+
+The `botanium` library will contain user-friendly bindings to specific algorithms, designed for ease of use much like `libsodium`, but using `botan` or `crypto-schemes-botan` as a backend.
+
+### botanite
+
+The `botanite` library will contain an as-close-as-possible reimplementation of `crypton/ite` using `botan` or `crypto-schemes-botan` as a back-end, and may be kept as a separate library or merged into `crypton` at a future time.
+
+## Goals and Non-goals
+
+The following are the immediate goals of this first leg:
+
+- To publish the `botan`, `botan-low`, and `botan-bindings` libraries
+- To build proper documentation and tutorials covering the subject material
+- To establish a community project presence focused on Haskell and cryptography
```

Comment:
I don't understand what success looks like for this goal. Can it be made more specific/actionable?

Reply:
Its less important now, I think this was from notes before I joined the haskell cryptography group, and the devlog has already succeeded in doing that anyway. It can be struck from this proposal